app/questionnaire/questionnaire_schema.py

```diff
@@ -290,6 +292,14 @@ def _get_blocks_by_id(self) -> dict[str, ImmutableDict]:
                             nested_block_id = nested_block["id"]
                             blocks[nested_block_id] = nested_block
                             self._parent_id_map[nested_block_id] = block_id
+                    if repeating_blocks := block.get("repeating_blocks"):
+                        for repeating_block in repeating_blocks:
+                            repeating_block_id = repeating_block["id"]
+                            blocks[repeating_block_id] = repeating_block
+                            self._parent_id_map[repeating_block_id] = block_id
+                            self._repeating_blocks_by_id[
```

Comment:
I think this can be a list of ids, the dict values are never used?

Reply:
I think these should be tracked as blocks by id like the other x_by_ids, as a means of collecting of them up front and making searching for the blocks by id more efficient

Reply:
Ah but then you can just use blocks_by_id for the block so double recording the repeating blocks is redundant -  yeah fair will change to only ids

Reply:
Done 👍